terminal/runereader.go

```diff
@@ -377,17 +378,24 @@ func (rr *RuneReader) ReadLineWithDefault(mask rune, d []rune, onRunes ...OnRune
 	}
 }
 
+// runeWidth returns the number of columns spanned by a rune when printed to the terminal
 func runeWidth(r rune) int {
 	switch width.LookupRune(r).Kind() {
 	case width.EastAsianWide, width.EastAsianFullwidth:
 		return 2
 	}
+
+	if !unicode.IsPrint(r) {
+		return 0
+	}
 	return 1
 }
 
+// StringWidth returns the visible width of a string when printed to the terminal
 func StringWidth(str string) int {
 	w := 0
-	rs := []rune(str)
+	cleanedStr := stripansi.Strip(str)
```

Comment:
I think it's an overkill to pull in a dependency just to get access to a single regular expression. However, since we're looping to measure the printable string width here, I would be more in favor of an approach like this one: https://github.com/muesli/reflow/blob/v0.3.0/ansi/buffer.go#L20-L40

I'm also not suggesting to import `github.com/muesli/reflow/ansi` here, but we could inline the approach of ignoring ANSI escape codes while iterating over a string. What do you think?

Reply:
I similarly agree that a whole import for this one line might be too much.. I also really like the iterative approach as it's slightly more readable than the regex too! I'll make these changes in the next day or two!